Clean up ReportingExtension and deprecate unnecessary methods (#34303)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_9.adoc

=== Deprecations

[[reporting_extension_file]]

> ==== `ReportingExtension.file(String)` has been deprecated

Reviewer: ```suggestion
==== Deprecation of `ReportingExtension.file(String)`
```

> This link:{javadocPath}/org/gradle/api/reporting/ReportingExtension.html#file(String)[method] has been deprecated and will be removed in Gradle 10.0.0.

Reviewer: ```suggestion
The link:{javadocPath}/org/gradle/api/reporting/ReportingExtension.html#file(String)[`file()` method] on `ReportingExtension` has been deprecated and will be removed in Gradle 10.0.0.
```

Use `ReportingExtension.getBaseDirectory()` with `file(String)` or `dir(String)` instead.

[[reporting_extension_api_doc_title]]

> ==== `ReportingExtension.getApiDocTitle()` has been deprecated

Reviewer: ```suggestion
==== Deprecation of `ReportingExtension.getApiDocTitle()`
```

> This link:{javadocPath}/org/gradle/api/reporting/ReportingExtension.html#getApiDocTitle()[method] has been deprecated and will be removed in Gradle 10.0.0. There is no direct replacement for this method.

Reviewer: ```suggestion
The link:{javadocPath}/org/gradle/api/reporting/ReportingExtension.html#getApiDocTitle()[`getApiDocTitle()` method] on `ReportingExtension` has been deprecated and will be removed in Gradle 10.0.0. 

There is no direct replacement for this method.
```
